### ISSUE TITLE
Segfault during install -- better debug options

### ISSUE THREAD

**Issue by mackeyja92** [sdispater/poetry#320]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: docker image `python:3.6-stretch`
- **Poetry version**: 0.11.2
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/mackeyja92/d541aea702e9c5636aad79b570724a50

The above gist includes the `Dockerfile`, `.gitlab-ci.yml`, `pyproject.toml`, and the log of the build running from our CI.

## Issue
We have been trying to hunt down the cause of our broken build for awhile now and have finally narrowed it down to installing `msgpack-python` on the `python:3.6-stretch` docker image while running docker in docker. That's a lot of variables. So the reason I am opening an issue is we could add some more debug options to poetry to help track these kinds of issues down. 

In summary: 
- Is the segfault even in poetry?
- Better debugging options around figuring out exactly which install segfaulted would be nice

**Comment by cauebs** [comment]
Well, the segfault occurred right after you got `- Installing msgpack-python (0.5.6)`, so it seems like the problem is in `msgpack-python`'s `setup.py`. I just checked and it uses Cython, so it makes sense that something could be going wrong.

What specific improvement would you suggest? We can't enable [faulthandler](https://docs.python.org/3/library/faulthandler.html) because it is not available on Python 2.7

**Comment by mackeyja92** [comment]
@cauebs well, this was after finding it. In our real project it happened at the end of the setup of many packages. I am not sure what other options exist to be able to suggest any. I do know that this has cost us many hours dev time to hunt down just which package was segfaulting. At one point we just went back to `requirements.txt` as it wasn't segfaulting.

**Comment by mackeyja92** [comment]
I guess I left out a critical piece of information, running `pip install msgpack-python` instead does not segfault in the build. So something that poetry is doing is causing it.

**Comment by mackeyja92** [comment]
@cauebs [here](https://gist.github.com/mackeyja92/f626e2747c7dd06e1db44c46716114bb) is another gist showing the install with `pip` and another install installing `asyncpg` which requires cython.

**Comment by mackeyja92** [comment]
@cauebs any status on this? Anything else I can supply to help?

**Comment by cauebs** [comment]
I just went back to the university and I simply don't have the time to investigate this right now. I'm sorry :(

**Comment by sdispater** [comment]
Closing. Feel free to reopen if the issue still exists.
